### ISSUE TITLE
[Bug] App Crash: IndexOutOfBoundsException

### ISSUE THREAD

**Issue by Ero-gamer** [T8RIN/ImageToolbox#1304]
Device: CPH2083 (OPPO - OP4BFB)
SDK: 28 (9)
App: 2.9.1-rc02 (151)java.lang.IndexOutOfBoundsException: off=9987102, len=93690 out of bounds (size=10000000)at java.nio.Buffer.checkBounds(Buffer.java:587)at java.nio.HeapByteBuffer.get(HeapByteBuffer.java:121)at H5.b.read(SourceFile:3)at com.github.penfeizhou.animation.io.b.read(Unknown Source:2)at J5.c.a(SourceFile:36)at J5.a.a(SourceFile:313)at ed.a.w(SourceFile:526)at S6.a.q(SourceFile:9)at y8.E0.z(SourceFile:54)at y8.a.q(SourceFile:23)at S6.a.q(SourceFile:32)at y8.H.run(SourceFile:95)at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)at java.lang.Thread.run(Thread.java:764)Suppressed: D8.g: [v0{Cancelling}@84290d8, java.util.concurrent.ThreadPoolExecutor@b89ac31[Running, pool size = 1, active threads = 1, queued tasks = 0, completed tasks = 259]]

**Comment by Ero-gamer** [comment]
App crash after selecting images frames (123 images) to be converted to webp animation and clicking convert button. Also GIF to WEBPA is broken, nothing happen after selecting gif file but app doesn't crash. Tried 3 types of image sources but all of them are the same option when selecting gif files so the same issue.

**Comment by T8RIN** [comment]
![image](https://github.com/user-attachments/assets/b651039f-78e0-408e-8e3a-f815666fa4ad)
i think this restriction didn't allow to do this, i will increase it

**Comment by T8RIN** [comment]
could you provide gif which isn't converting? Because when i tried it worked perfectly

**Comment by Ero-gamer** [comment]
> could you provide gif which isn't converting? Because when i tried it worked perfectly

The issue is not in gif file itself but rather file picker that for some reason doesn't work on my A9. Selecting gif files through file managers file picker result in nothing and switch back to app GIF tools screen. Tried all different file managers file picker and various different gif files but same issue.

**Comment by T8RIN** [comment]
Use default picker, it works fine 👀

**Comment by Ero-gamer** [comment]
> Use default picker, it works fine 👀

As I write above, I have tried all file pickers including default file picker but same issue. Maybe it's because I debloated my A9 phone that may deleted or effected default file picker? BTW default file picker works in format conversion but doesn't work when selecting gif, it only effect gif file selection. Is it possible to implement gallery file picker when selecting gif file? This might solve the problem because gallery picker is not available when choosing gif files even when you set it in image source settings.
